multicluster/controllers/multicluster/member/gateway_controller.go

```diff
@@ -84,17 +82,12 @@ func NewGatewayReconciler(
 //+kubebuilder:rbac:groups=multicluster.crd.antrea.io,resources=clusterinfoimports/finalizers,verbs=update
 
 func (r *GatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	if !r.checkClusterSetReady() {
-		klog.InfoS("Skip reconciling Gateway since there is no connection to the leader")
-		return ctrl.Result{}, nil
-	}
-
 	klog.V(2).InfoS("Reconciling Gateway", "gateway", req.NamespacedName)
-	var err error
 	var commonArea commonarea.RemoteCommonArea
-	commonArea, r.localClusterID, err = r.commonAreaGetter.GetRemoteCommonAreaAndLocalID()
+	commonArea, r.localClusterID, _ = r.commonAreaGetter.GetRemoteCommonAreaAndLocalID()
```

Comment:
There can be a case the commonArea is to be stopped after cleanup is done (which may be in progress or failed). Do we have a way to check that?

Reply:
I don't think that will be a case, GetRemoteCommonAreaAndLocalID() is getting commonArea via r.commonAreaGetter from ClusterSet controller, the commonArea is protected by commonAreaLock. If Reconcile in ClusterSet controller is processing the ClusterSet change, it will lock the commonArea, then `GetRemoteCommonAreaAndLocalID()` won't be able to get one before it gets the read lock.

Reply:
I meant the cleanup retry case (before retry happens), but maybe that is not critical to cover.

I now feel a worse thing is there can be a race condition between these controllers and memberAnnounce deletion in ClusterSet controller, so they can for example create more ResourceExports even after memberAnnounce is deleted. I do not have a simple idea to solve that, and if you do not either, probably let us add a comment here and address the issue later.